fpga_interchange/examples/chipdb.cmake

```diff
@@ -0,0 +1,187 @@
+function(create_rapidwright_device_db)
+    # ~~~
+    # create_rapidwright_device_db(
+    #    device <common device>
+    #    part <part>
+    # )
+    # ~~~
+    #
+    # Generates a device database from RapidWright
+    #
+    # Targets generated:
+    #   - rapidwright-<device>-device
+
+    set(options)
+    set(oneValueArgs device part)
+    set(multiValueArgs)
+
+    cmake_parse_arguments(
+        create_rapidwright_device_db
+        "${options}"
+        "${oneValueArgs}"
+        "${multiValueArgs}"
+        ${ARGN}
+    )
+
+    set(device ${create_rapidwright_device_db_device})
+    set(part ${create_rapidwright_device_db_part})
+    set(rapidwright_device_db ${CMAKE_CURRENT_BINARY_DIR}/${part}.device)
+    add_custom_command(
+        OUTPUT ${rapidwright_device_db}
+        COMMAND
+            RAPIDWRIGHT_PATH=${RAPIDWRIGHT_PATH}
+            ${INVOKE_RAPIDWRIGHT}
+            com.xilinx.rapidwright.interchange.DeviceResourcesExample
+            ${part}
+        DEPENDS
+            ${INVOKE_RAPIDWRIGHT}
+    )
+
+    add_custom_target(rapidwright-${device}-device DEPENDS ${rapidwright_device_db})
```

Comment:
There needs to an option to see the output target name to a variable in parent scope.

Reply:
Done